tyaml/dumper.py

```diff
@@ -16,4 +18,42 @@ def represent_object(self, data):
         return self.represent_dict(result_dict)
 
 
-SpecialDumper.add_multi_representer(object, SpecialDumper.represent_object)
+OutputOptionalNullsDumper.add_multi_representer(object, OutputOptionalNullsDumper.represent_object)
+
+class NoOutputOptionalNullsDumper(Dumper):  # pylint: disable=too-many-ancestors
+    """Dumper overriding standard `represent_object` method"""
+
+    def represent_object(self, data):
+        mappings = get_mappings(type(data))  # find yaml fields mappings
+        if not mappings:
+            _data = data
+            try:
+                _data = vars(data)
+                for key in _data.keys():
+                    if self._is_optional(data, key):
```

Comment:
This check is running on the instance. Instance attributes can't have the type `Optional`, it will either be some real type or None. You should check `type(data)` fields, same as it is done for mappings.